src/loaders/TiffLoader.js

```diff
@@ -0,0 +1,252 @@
+import { isInTileBounds } from './utils';
+
+/**
+ * This class serves as a wrapper for reading tiff data.
+ * @param {Object} args
+ * @param {Object} args.tiff geotiffjs tiff object.
+ * @param {Object} args.pool Pool that implements a `decode` function.
+ * @param {Object} args.firstImage First image (geotiff Image object) in the tiff (containing base-resolution data).
+ * @param {Object} args.dimensions Dimensions for the loader.
+ * @param {Object} args.metadata The metadata object
+ * @param {Array} args.offsets The offsets of each IFD in the tiff.
+ * @param {Boolean} args.isRgb Whether or not this tiff represents an rgb image.
+ * @param {Boolean} args.isInterleaved Whether or not this tiff represents an interleaved image.
```

Comment:
General curiosity: what is the difference between these two? it seems like if an image is interleaved we can assume it is RGB/A (e.g. needs our BitmapLayer). If it isn't interleaved, then its up to the UI to determine colors to apply to each channel, no? So ultimately, `isRgb` here is conveying something about the rendering metadata and how `colorValues` prop should be set?